pippy/PipelineStage.py

```diff
@@ -85,9 +85,12 @@ def __init__(
             self.global_depth = nstages
             self.inner_depth = 1
 
+        self.pipe_cache = [{} for i in range(chunks)] # [chunk][inner_rank] = value
+
         # Find my submodule
         self.split_gm = self.pipe.split_gm
         named_children = list(self.split_gm.named_children())
+
```

Comment:
is this still needed?

Reply:
@fduwjj  comment is left on blank line. if it indicate L91-L92 or L95, we will leave it for safety reason until we modify all functions to use 'list of submods' instead of one submod.

Reply:
I am indeed asking about L95.